central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -124,11 +131,80 @@ func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *ComplianceRep
 	go rg.sendEmail(req.Ctx, zipData, formatEmailBody, formatEmailSub, req.Notifiers)
 }
 
-// getDataForReport returns map of cluster id and
-func (rg *complianceReportGeneratorImpl) getDataForReport(req *ComplianceReportRequest) (*ResultEmail, error) {
-	// TODO ROX-24356: Implement query to get checkresults data to generate cvs for compliance reporting
+// getDataforReport returns map of cluster id and
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest) (*ResultEmail, error) {
+
+	clusters := req.ClusterIDs
+
+	resultsCSV := make(map[string][]*ResultRow)
+
+	resultEmailComplianceReport := &ResultEmail{
+		TotalPass:  0,
+		TotalMixed: 0,
+		TotalFail:  0,
+		Clusters:   0,
+	}
+
+	for _, clusterID := range clusters {
+
+		parsedQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfig, req.ScanConfigID).
+			AddExactMatches(search.ClusterID, clusterID).
+			ProtoQuery()
+		resultCluster := []*ResultRow{}
+
+		err := rg.checkResultsDS.WalkByQuery(req.Ctx, parsedQuery, func(c *storage.ComplianceOperatorCheckResultV2) error {
+			row := &ResultRow{
+				ClusterName: c.GetClusterName(),
+				CheckName:   c.GetCheckName(),
+				Description: c.GetDescription(),
+				Status:      c.GetStatus().String(),
+				Remediation: c.GetInstructions(),
+			}
+			// get profile name for the check result
+			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, c.GetScanRefId()).ProtoQuery()
+			scans, err := rg.scanDS.SearchScans(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(scans) < 1 {
+				return errors.New("Rule not found")
+			}
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorProfileID, scans[0].GetProfile().GetProfileId()).ProtoQuery()
+			profiles, err := rg.profileDS.SearchProfiles(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(profiles) < 1 {
+				return errors.New("Profile not found")
+			}
+			row.Profile = profiles[0].GetName()
```

Comment:
We should be able to do it like this.  I have been doing it this way in results.  The search framework figures out the joins.
```suggestion
			// get profile name for the check result
			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, c.GetScanRefId()).ProtoQuery()
			profiles, err := rg.profileDS.SearchProfiles(req.Ctx, q)
			if err != nil {
				return err
			}
			if len(profiles) < 1 {
				return errors.New("Profile not found")
			}
			row.Profile = profiles[0].GetName()
```